maven-enforcer-plugin/src/test/java/org/apache/maven/plugins/enforcer/TestEnforceMojo.java

```diff
@@ -296,6 +296,14 @@ public void testLoggingOnEnforcerRuleExceptionWithoutMessage()
         Mockito.verify( logSpy ).warn( Mockito.matches( ".* failed with message:" + System.lineSeparator() + "null" ) );
     }
 
+    @Test
+    public void testFailIfNoTests()
+            throws MojoExecutionException {
+        setupBasics( false );
+        mojo.setFailIfNoRules( false );
+        mojo.execute();
+    }
```

Comment:
please add some assertions, eg that `log.warn` was called only once with specific message and no more interactions on `log`